pooch/tests/test_core.py

```diff
@@ -74,21 +75,45 @@ def test_pooch_download():
         true_path = str(path / "tiny-data.txt")
         # Setup a pooch in a temp dir
         pup = Pooch(path=path, base_url=BASEURL, registry=REGISTRY)
-        # Check that the warning says that the file is being downloaded
-        with warnings.catch_warnings(record=True) as warn:
+        # Check that the logs say that the file is being downloaded
+        with capture_log() as log_file:
             fname = pup.fetch("tiny-data.txt")
-            assert len(warn) == 1
-            assert issubclass(warn[-1].category, UserWarning)
-            assert str(warn[-1].message).split()[0] == "Downloading"
-            assert str(warn[-1].message).split()[-1] == "'{}'.".format(path)
+            logs = log_file.getvalue()
+            assert logs.split()[0] == "Downloading"
+            assert logs.split()[-1] == "'{}'.".format(path)
         # Check that the downloaded file has the right content
         assert true_path == fname
         check_tiny_data(fname)
         assert file_hash(fname) == REGISTRY["tiny-data.txt"]
-        # Check that no warnings happen when not downloading
-        with warnings.catch_warnings(record=True) as warn:
+        # Check that no logging happens when not downloading
+        with capture_log() as log_file:
             fname = pup.fetch("tiny-data.txt")
-            assert not warn
+            assert log_file.getvalue() == ""
+
+
+def test_pooch_logging_level():
+    "Setup a pooch and check that no logging happens when the level is raised"
+    with TemporaryDirectory() as local_store:
+        # Add a new handler for critical events only
+        critical_log_file = StringIO()
+        critical_handler = StreamHandler(critical_log_file)
+        critical_handler.setLevel("CRITICAL")
+        get_logger().addHandler(critical_handler)
+        path = Path(local_store)
+        urls = {"tiny-data.txt": BASEURL + "tiny-data.txt"}
+        # Setup a pooch in a temp dir
+        pup = Pooch(path=path, base_url="", registry=REGISTRY, urls=urls)
+        # Check that the logs say that the file is being downloaded
+        with capture_log() as log_file:
```

Comment:
Would it possible to pass in a `level` argument to `capture_log` and have it set the level on the handler?

Reply:
Very possible and much better, thanks!